src/main/scala/sparkz/core/network/NodeViewSynchronizer.scala

```diff
@@ -325,13 +325,12 @@ class NodeViewSynchronizer[TX <: Transaction, SI <: SyncInfo, SIS <: SyncInfoMes
     modifiers.flatMap { case (id, bytes) =>
       val reader: VLQReader = new VLQByteBufferReader(ByteBuffer.wrap(bytes))
 
-      if (reader.remaining != 0) {
-        penalizeMisbehavingPeer(remote)
-        log.warn(s"Received additional bytes after block. Declared id ${encoder.encodeId(id)} from ${remote.toString}")
-      }
-
       serializer.parseTry(reader) match {
         case Success(mod) if id == mod.id =>
+          if (reader.remaining != 0) {
```

Comment:
Let's have some UTs on Sparkz level. Don't want to keep it untested within the Sparkz project, but only on SDK side.
Note: you may define a dummy tx or block serializers for this.

Reply:
Added UT covering these cases